File: Leetcode/Daily/06-2025/TypeScript/21.ts
// Problem: 3085. Minimum Deletions to Make String K-Special (Medium)
// Link: https://leetcode.com/problems/minimum-deletions-to-make-string-k-special

function minimumDeletions(word: string, k: number): number {
    let freq = new Array(26).fill(0);
    
    for (let ch of word) {
        freq[ch.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }
    
    freq.sort((a, b) => a-b);
    let ans = Number.MAX_SAFE_INTEGER;
    
    for (let i = 25; i >= 0; i--) {
        let temp = 0;
        for (let j = i-1; j >= 0; j--) {
            temp += freq[j];
        }
        let target = freq[i] + k;
        for (let j = i; j < 26; j++) {
            if (freq[j] <= target) continue;
            temp += freq[j] - target;
        }
        ans = Math.min(ans, temp);
    }

    return ans;            
};

File: Leetcode/Dailies/2025/01/21.cpp
// Problem: 2017. Grid Game (Medium)
// Link: https://leetcode.com/problems/grid-game

class Solution {
public:
    long long gridGame(vector<vector<int>>& grid) {
        int n = grid[0].size();
        long long a = 0, b = 0, ans = LLONG_MAX;

        for (int i = 0; i < n; i++) a += grid[0][i];

        for (int i = 0; i < n; i++) {
            a -= grid[0][i];

            ans = min(ans, max(a, b));

            b += grid[1][i];
        }

        return ans;
    }
};